internal/backends/sqlite/collection.go

```diff
@@ -164,9 +174,35 @@ func (c *collection) Update(ctx context.Context, params *backends.UpdateParams)
 
 // Delete implements backends.Collection interface.
 func (c *collection) Delete(ctx context.Context, params *backends.DeleteParams) (*backends.DeleteResult, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/2751
+	db := c.r.DatabaseGetExisting(ctx, c.dbName)
+	if db == nil {
+		return nil, lazyerrors.Errorf("no database %q", c.dbName)
+	}
+
+	tableName := c.r.CollectionToTable(c.name)
+
+	query := fmt.Sprintf(`DELETE FROM %q WHERE json_extract(_ferretdb_sjson, '$._id') = ?`, tableName)
+
+	var deleted int64
+
+	for _, id := range params.IDs {
+		idArg := strings.ReplaceAll(string(must.NotFail(sjson.MarshalSingleValue(id))), "\"", "")
```

Comment:
The same for update

Reply:
I still don't know the reason behind but it would not work with `"`. The update tests were failing because of it too. Will look into it but suggest proceeding with the current solution.

Reply:
That's because `sjson.MarshalSingleValue(id)` produces []byte and using `string()` on it produces `""int64-min""`. Maybe psql needs its specific method other than `sjson.MarshalSingleValue`?

Perhaps we need tests where IDs are all supported types other than string. Maybe we have such tests already? This can of course be a separate issue.

Reply:
> but suggest proceeding with the current solution

No, we don't propagate code we don't understand.

> That's because sjson.MarshalSingleValue(id) produces []byte and using string() on it produces ""int64-min""

Converting `[]byte` to `string` or back does not magically add quotes: https://go.dev/play/p/4y3lMXGJydk

Reply:
I changed the update and delete functions to use `->` and it worked.

Reply:
Perfect!